Python/Python-tokenize.c

```diff
@@ -175,14 +176,64 @@ _tokenizer_error(struct tok_state *tok)
     return result;
 }
 
+static PyObject *
+_get_current_line(tokenizeriterobject *it, const char *line_start, Py_ssize_t size)
+{
```

Comment:
You might want to add a macro that asserts `it` is locked See `_Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED` or `ASSERT_DICT_LOCKED` for a macro redefinition of that.